### PR TITLE
Homepage layout + type adjusments

### PR DESCRIPTION
**Pull request by emilybrick**
Closes https://github.com/github/primer/issues/2428

Screen:
Screenshot 2023-07-21 at 12 07 34 PM
Screenshot 2023-07-21 at 12 07 27 PM

### CODE REVIEW

File: content/index.mdx [hunk 1]
```diff
@@ -4,27 +4,53 @@ title: Interface guidelines
...
-
-Design guidelines covering common user workflows.
-
-#### [Components](/components)
-
-Design and usage guidelines for individual Primer components.
-
-## Need help?
-
-If you found a bug on this website, please [open a new issue](https://github.com/primer/design/issues/new) with as much detail as possible, including steps to replicate the bug, screenshots, links, and expected results.
```
Comment (emilybrick) [inline content/index.mdx:28]: Not sure where this should be 🤔

Comment (josepmartins) [reply content/index.mdx:28]: Seems related to the open source section 🤔 

Screenshot 2023-07-21 at 09 55 44

File: content/components/profile-button.mdx [hunk 2]
```diff
@@ -0,0 +1,15 @@
+---
```
Comment (mperrotti) [inline content/components/profile-button.mdx:1]: What is this?

File: content/index.mdx [hunk 3]
```diff
@@ -4,27 +4,53 @@ title: Interface guidelines
...
+import {Link as GatsbyLink} from 'gatsby'
+
+<Box mb={6}>
+  <Text as="p" sx={{m: 0, fontSize: 3}}>
+    Primer is a set of guidelines, principles, and patterns for designing and building UI at GitHub. It provides a shared language and standardized approach to delivering cohesive experiences.
+  </Text>
+</Box>
+
+<Box display="grid" gridTemplateColumns={['1fr', null, null, null, '1fr 1fr']} gridGap={5}>
+  <Box>
```
Comment (mperrotti) [inline content/index.mdx:17]: Non-blocking nitpick: These can probably just be a regular `<div>`

File: content/index.mdx [hunk 4]
```diff
@@ -4,27 +4,53 @@ title: Interface guidelines
...
+    <Heading as="h3" sx={{fontSize: 3}}><Link sx={{fontSize: 3,}} as={GatsbyLink} to="/guides/">Guides</Link></Heading>
+    <Box as="p" sx={{fontSize: 2, color: 'fg.muted'}}>Standards, guidelines, and tools to getting started with Primer.</Box>
+  </Box>
+  <Box>
+    <Heading as="h3" sx={{fontSize: 3}}><Link sx={{fontSize: 3,}} as={GatsbyLink} to="/foundations/">Foundations</Link></Heading>
+    <Box as="p" sx={{fontSize: 2, color: 'fg.muted'}}>The fundamental parts of the design system that underpin all GitHub interfaces, such as color and typography.</Box>
+  </Box>
```
Comment (mperrotti) [inline content/index.mdx:24]: Non-blocking nitpick: this could probably just be a component so we don't have to repeat ourselves so much.

**Review by mperrotti — COMMENTED** [review]
LGTM! Will approve once I understand what `profile-button.mdx` is for.

**Review by josepmartins — APPROVED** [review]
LGTM!

**Pull request merged**